agent/agents/mongodb/internal/profiler/aggregator/extended.go

```diff
@@ -0,0 +1,270 @@
+	}
+
+	s.Reset()
+	return s
+}
+
+// Reset clears the collection of statistics
+func (s *extendedStats) Reset() {
+	s.Lock()
+	defer s.Unlock()
+
+	s.queryInfoAndCounters = make(map[stats.GroupKey]*extendedQueryInfoAndCounters)
+}
+
+// Queries returns all collected statistics
+func (s *extendedStats) Queries() extendedQueries {
+	s.Lock()
+	defer s.Unlock()
+
+	keys := stats.GroupKeys{}
+	for key := range s.queryInfoAndCounters {
+		keys = append(keys, key)
+	}
+	sort.Sort(keys)
+
+	queries := []extendedQueryInfoAndCounters{}
+	for _, key := range keys {
+		queries = append(queries, *s.queryInfoAndCounters[key])
+	}
+	return queries
+}
+
+// Add adds collector.ExtendedSystemProfile to the collection of statistics
+func (s *extendedStats) Add(doc collector.ExtendedSystemProfile) error {
+	fp, err := s.fingerprinter.Fingerprint(doc.SystemProfile)
+	if err != nil {
+		return err // TODO &stats.StatsFingerprintError{err}
+	}
+	var qiac *extendedQueryInfoAndCounters
+	var ok bool
+
+	key := stats.GroupKey{
+		Operation:   fp.Operation,
+		Fingerprint: fp.Fingerprint,
+		Namespace:   fp.Namespace,
+	}
+	if qiac, ok = s.getQueryInfoAndCounters(key); !ok {
+		query := proto.NewExampleQuery(doc.SystemProfile)
+		queryBson, err := bson.MarshalExtJSON(query, true, true)
+		if err != nil {
+			return err
+		}
+		qiac = &extendedQueryInfoAndCounters{
+			QueryInfoAndCounters: stats.QueryInfoAndCounters{
+				ID:          fmt.Sprintf("%x", md5.Sum([]byte(key.String()))),
+				Operation:   fp.Operation,
+				Fingerprint: fp.Fingerprint,
+				Namespace:   fp.Namespace,
+				TableScan:   false,
+				Query:       string(queryBson),
+			},
+			PlanSummary: doc.PlanSummary,
+		}
+		s.setQueryInfoAndCounters(key, qiac)
+	}
+	qiac.Count++
+	qiac.PlanSummary = doc.PlanSummary
+	if qiac.PlanSummary == planSummaryCollScan {
+		qiac.CollScanCount++
+		qiac.CollScanSum += int64(doc.Millis)
+	}
+	if strings.HasPrefix(qiac.PlanSummary, planSummaryIXScan) {
+		qiac.PlanSummary = planSummaryIXScan
+	}
+	// docsExamined is renamed from nscannedObjects in 3.2.0.
+	// https://docs.mongodb.com/manual/reference/database-profiler/#system.profile.docsExamined
+	s.Lock()
+	if doc.NscannedObjects > 0 {
+		qiac.NScanned = append(qiac.NScanned, float64(doc.NscannedObjects))
+	} else {
+		qiac.NScanned = append(qiac.NScanned, float64(doc.DocsExamined))
+	}
+	qiac.NReturned = append(qiac.NReturned, float64(doc.Nreturned))
+	qiac.QueryTime = append(qiac.QueryTime, float64(doc.Millis))
+	qiac.ResponseLength = append(qiac.ResponseLength, float64(doc.ResponseLength))
+	if qiac.FirstSeen.IsZero() || qiac.FirstSeen.After(doc.Ts) {
+		qiac.FirstSeen = doc.Ts
+	}
+	if qiac.LastSeen.IsZero() || qiac.LastSeen.Before(doc.Ts) {
+		qiac.LastSeen = doc.Ts
+	}
+	s.Unlock()
+
+	return nil
+}
+
+func (s *extendedStats) getQueryInfoAndCounters(key stats.GroupKey) (*extendedQueryInfoAndCounters, bool) {
+	s.RLock()
+	defer s.RUnlock()
+
+	v, ok := s.queryInfoAndCounters[key]
+	return v, ok
```

Comment:
```suggestion
	return s.queryInfoAndCounters[key]
```

Reply:
@ademidoff I just reverted suggested change. With your approach it wont return ok, but just only value.

Reply:
Yes, I saw it :) 
Nevermind